reactive/rerunner.go

```diff
@@ -350,23 +350,27 @@ func (r *Rerunner) run() {
 	ctx := context.WithValue(r.ctx, cacheKey{}, r.cache)
 	ctx = context.WithValue(ctx, dependencySetKey{}, &dependencySet{})
 
-	computation, err := run(ctx, r.f)
+	currentComputation, err := run(ctx, r.f)
 	r.lastRun = time.Now()
 	if err != nil {
-		if err == RetrySentinelError {
-
-			r.retryDelay = r.retryDelay * 2
-
-			// Max out the retry delay to at 1 minute.
-			if r.retryDelay > time.Minute {
-				r.retryDelay = time.Minute
-			}
-			go r.run()
-		} else {
+		if err != RetrySentinelError {
 			// If we encountered an error that is not the retry sentinel,
 			// we should stop the rerunner.
 			return
 		}
+		// Reset the cache for sentinel errors so we get a clean slate.
+		r.cache = &cache{
+			computations: make(map[interface{}]*computation),
+			locker:       newLocker(),
+		}
```

Comment:
maybe need a `go oldCache.CleanupCache()` ?

Reply:
i think it'll jsut get GCd